### PR TITLE
Js daily notes feature

### PR DESCRIPTION
**Pull request by johnschoeman**
Relatively simple feature, nothing too crazy, but i did have two questions that would be nice to chat about irl.

### Q1: 
what are thoughtbot's thoughts on before_action filters in controllers?

I looked but could only find this: https://robots.thoughtbot.com/before-filter-wisdom which doesn't seem to have an opinion.  The reason I asks is because after watching this upcase video: 

https://thoughtbot.com/upcase/videos/thoughtbots-approach-to-ruby

there is a lot of discussion about making code not hide complexity by using fancy ruby tricks, like ternaries, or end of line if statements.  Wouldn't before_filters in controllers fall into category as well?  Though using them allows the controller to tell a more easily followable story.

---

### Q2:
would it make more sense to only have the notes#update action and initialize a note on entry creation?
or to have an notes#create action and a notes#update action and have a conditional in the controller to choose which?

I like having the notes controller be responsible for both creation and updating of notes, because it feels like breaking single responsibility to have the entries or entry_form objects to handle creation of notes. (unless we allow notes on the new/edit entry forms, which i don't want from a ux perspective) Though doing this might save us from having nil checks and conditionals in the views/controllers, since the note will always exist.

another option would be to have the create action delete the previous note and have it act as a update, (also don't like this due to obfuscation of what create means)

### CODE REVIEW

File: app/assets/stylesheets/entry.scss [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.entry-notes {
+  min-height: 400px;
+  width: 60%;
+}
```
Comment (christoomey) [inline app/assets/stylesheets/entry.scss]: As a general habit, I'd recommend including a screenshot or gif when opening a PR with any new UI or workflows. Certainly not required in this case, but a nice to have and a good habit to get into.

File: app/controllers/notes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  def require_entry_to_belong_to_user
+    entry = entry_for_notes
+    if entry.user == current_user
+      entry
+    else
+      flash[:error] = "Entry does not belong to you"
+      redirect_to entries_path
+    end
+  end
```
Comment (christoomey) [inline app/controllers/notes_controller.rb]:
``` ruby
def require_entry_to_belong_to_user
  unless entry_for_notes.user == current_user
    redirect_to entries_path, error: "Entry does not belong to you"
  end
end
```

1. We're use `before_action`s for access control, as you've used it 👍. We strongly avoid it for things like assigning instance variables, e.g. `before_action :lookup_entry` or similar.
2. I think (not sure) that you can combine the `redirect_to` and the error flash as I've shown
3. No need to return the entry (unless I'm forgetting something about returning `nil`?)

File: app/controllers/notes_controller.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  private
+
+  def note_params
+    params.require(:note).permit(:content)
+  end
+
+  def entry_for_notes
+    @_entry_for_notes ||= Entry.find(params[:entry_id])
```
[the remark this answers is not in the document]
Comment (christoomey) [reply app/controllers/notes_controller.rb]: I'd be OK with calling this the `def entry`.

File: app/controllers/notes_controller.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  def create
+    @entry = entry_for_notes
+    note = Note.new(content: note_params[:content], entry: @entry)
+
+    if note.save
+      flash[:success] = "Note Saved"
+    else
+      flash[:error] = note.errors.full_messages
+    end
+    redirect_to entry_path(@entry)
```
Comment (christoomey) [inline app/controllers/notes_controller.rb]: Is there a way you can combine these two sections that are mostly common between the `create` and `update` actions?

File: app/views/entries/_note_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  <% action_url = entry_note_path(entry, note) %>
+<% end %>
+
+<%= form_for note, url: action_url do |note_form| %>
+  <label for="entry-notes">Notes</label>
+  <textarea
+    id="entry-notes"
+    class="entry-notes"
+    name="note[content]"><%= entry.note.content %></textarea>
+  <input type="submit" value="Save Notes" />
```
Comment (christoomey) [inline app/views/entries/_note_form.html.erb]: I'd suggest taking a stab at using the form helpers to generate these fields. `form.text_area :notes` and similar.

File: app/views/entries/show.html.erb [hunk 6]
```diff
@@ -10,6 +10,8 @@
 <% end %>
 </ul>
 
+<%= render "note_form", note: @note, entry: @entry %>
```
Comment (christoomey) [inline app/views/entries/show.html.erb:13]: 👍 to the extraction, but explicitly passing in the instance variables.

**Comment by johnschoeman** [comment]
@christoomey didn't get an explicit LGTM from you.  but comments have been addressed.  is this GTG?  Thanks!

**Pull request merged**